File: app/src/main/java/com/boardgamegeek/provider/GamesIdPlaysProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Plays;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesIdPlaysProvider extends BaseProvider {

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		String gameId = uri.getLastPathSegment();
		return new SelectionBuilder()
			.table(Tables.PLAYS)
			.whereEquals(Plays.OBJECT_ID, gameId);
	}

	@Override
	protected String getDefaultSortOrder() {
		return Plays.DEFAULT_SORT;
	}

	@Override
	protected String getPath() {
		return "games/#/plays";
	}

	@Override
	protected String getType(Uri uri) {
		return Plays.CONTENT_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesIdPlaysProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggContract.Companion.PATH_PLAYS
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesIdPlaysProvider : BaseProvider() {
    override fun getType(uri: Uri) = Plays.CONTENT_TYPE

    override val path = "$PATH_GAMES/#/$PATH_PLAYS"

    override val defaultSortOrder = Plays.DEFAULT_SORT

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val gameId = Games.getGameId(uri)
        return SelectionBuilder()
            .table(Tables.PLAYS)
            .whereEquals(Plays.Columns.OBJECT_ID, gameId)
    }
}
